tests/src/JIT/Regression/JitBlue/GitHub_19550/GitHub_19550.cs

```diff
@@ -25,24 +25,77 @@ struct S
         public float f8;
         public float f9;
         public float f10;
+        public float f11;
+        public float f12;
+        public float f13;
+        public float f14;
+        public float f15;
     }
 
     [MethodImpl(MethodImplOptions.NoInlining)]
     static unsafe int Test(ref S s, Vector128<float> v, int offset)
     {
         int returnVal = 100;
-        // offset must be a multiple of the vector size in floats.
-        offset &= ~3;
+
         fixed (float* p = &s.f0)
         {
-            // We need an aligned address.
-            int alignmentOffset = (((int)p & 0xc) >> 2);
+            // We need an address aligned on 16 bytes, so we need to add a *float* offset to get there.
+            int alignmentOffset = (0x10 - ((int)p & 0xc)) >> 2;
             try
             {
                 // This is the aligned case.
+                // We're going to store a scalar at an offset of 2 from the aligned location.
+                // As it happens, we know that the struct has been initialized to all zeros,
+                // and the vector passed in was all ones, so now we have a one at offset 2.
                 Sse2.StoreScalar(p + alignmentOffset + 2, Sse2.Subtract(v, Sse2.LoadAlignedVector128(p + offset + alignmentOffset + 4)));
-                // This is the unaligned case.
-                Sse2.StoreScalar(p + alignmentOffset + 1, Sse2.Subtract(v, Sse2.LoadVector128(p + offset + alignmentOffset + 1)));
+
+                // Now do a non-temporal load from the aligned location.
+                // That should give us {0, 0, 1, 0}.
+                Vector128<float> v2 = Sse41.LoadAlignedVector128NonTemporal((byte*)(p + alignmentOffset)).AsSingle();
```

Comment:
`Sse41.IsSupported`?

Reply:
Right - of course! Forgot all about the checks!